jdaviz/core/marks.py

```diff
@@ -105,7 +106,7 @@ def set_y_unit(self, unit=None):
             unit = self.viewer.state.y_display_unit
         unit = u.Unit(unit)
 
-        if self.yunit is not None:
+        if self.yunit is not None and not np.all([s == 0 for s in self.y.shape]):
```

Comment:
I wasn't sure if this would ever apply to a multi-dimensional flux so I figured I'd be safe.